src/js/relay.firefox.com/get_profile_data.js

```diff
@@ -1,419 +1,453 @@
+          }
 
-        aliasLabelInput.dataset.label = newAliasLabel;
-        setTimeout(() => {
-          aliasLabelWrapper.classList.remove("show-saved-confirmation");
-        }, 1000);
-      };
-      aliasLabelInput.addEventListener("focusout", () => {
-        const isValid = aliasLabelForm.reportValidity();
-        if (isValid) {
+          // Save new alias label
+          const updatedRelayAddress = localStorageRelayAddresses.filter(
+            (address) => address.id == aliasId
+          )[0];
+          updatedRelayAddress.description = newAliasLabel;
+          browser.storage.local.set({
+            relayAddresses: localStorageRelayAddresses,
+          });
+
+          // show placeholder text if the label is blank
+          if (aliasLabelInput.value === "") {
+            aliasLabelWrapper.classList.remove("user-created-label");
+            aliasLabelInput.placeholder = defaultAliasLabelText;
+          } else {
+            aliasLabelWrapper.classList.add("user-created-label");
+            aliasLabelWrapper.classList.add("show-saved-confirmation");
+          }
+
+          aliasLabelInput.dataset.label = newAliasLabel;
+          setTimeout(() => {
+            aliasLabelWrapper.classList.remove("show-saved-confirmation");
+          }, 1000);
+        };
+        aliasLabelInput.addEventListener("focusout", () => {
+          const isValid = aliasLabelForm.reportValidity();
+          if (isValid) {
+            saveAliasLabel();
+          }
+        });
+        aliasLabelForm?.addEventListener("submit", (event) => {
+          event.preventDefault();
           saveAliasLabel();
-        }
-      });
-      aliasLabelForm?.addEventListener("submit", (event) => {
-        event.preventDefault();
-        saveAliasLabel();
-        aliasLabelInput.blur();
-      });
+          aliasLabelInput.blur();
+        });
 
-      // Get and store the relay addresses from the account profile page,
-      // so they can be used later, even if the API endpoint is down
+        // Get and store the relay addresses from the account profile page,
+        // so they can be used later, even if the API endpoint is down
 
-      const relayAddress = {
-        id: aliasId,
-        address: aliasCardData.relayAddress,
-        description: storedAliasLabel,
-        generated_for: storedAliasGeneratedFor,
-      };
+        const relayAddress = {
+          id: aliasId,
+          address: aliasCardData.relayAddress,
+          description: storedAliasLabel,
+          generated_for: storedAliasGeneratedFor,
+        };
 
-      localStorageRelayAddresses.push(relayAddress);
+        localStorageRelayAddresses.push(relayAddress);
+      }
+
+      if (localStorageRelayAddresses.length === 0) {
+        // If we weren't able to scrape alias data from the page, that means
+        // the React version of the website (with a different DOM structure)
+        // has been deployed. Since the React version handles local storage of
```

Comment:
Good question. So `localStorageRelayAddress` gets populated in this for loop:

https://github.com/mozilla/fx-private-relay-add-on/blob/e099130eaf5cc8ce66c74c6cb127d008c4ba2b23/src/js/relay.firefox.com/get_profile_data.js#L228

That's looping over all elements matching `[data-relay-address]`:

https://github.com/mozilla/fx-private-relay-add-on/blob/e099130eaf5cc8ce66c74c6cb127d008c4ba2b23/src/js/relay.firefox.com/get_profile_data.js#L108-L111

That will work in the current site. In the new site, however, there are no elements matching that query selector, so the loop will never run. (This is fine, because it does just two things: fetch alias data, and set up local label storage. As I'll mention in the other comment, the add-on will no longer be responsible for local label storage. And this `if` block takes over the fetching of alias data.)

It does mean that in the current site, if the user has 0 aliases, the add-on will also do an API request to again fetch those zero aliases. That seems acceptable in the transitory period, but open to reconsider if you disagree.